internal/pkg/consensus/election.go

```diff
@@ -1,37 +1,38 @@
 package consensus
 
 import (
-	"encoding/binary"
+	"context"
 	"math/big"
 
 	"github.com/filecoin-project/go-address"
 	sector "github.com/filecoin-project/go-sectorbuilder"
 	"github.com/filecoin-project/specs-actors/actors/abi"
+	acrypto "github.com/filecoin-project/specs-actors/actors/crypto"
+	"github.com/pkg/errors"
 
 	ffi "github.com/filecoin-project/filecoin-ffi"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/block"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/crypto"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/encoding"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/postgenerator"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/proofs/verification"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/types"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/util/convert"
 )
 
-// ElectionMachine generates and validates PoSt partial tickets and PoSt
-// proofs.
-type ElectionMachine struct{}
+// ElectionMachine generates and validates PoSt partial tickets and PoSt proofs.
+type ElectionMachine struct {
+	chain ChainRandomness
```

Comment:
The huge argument lists were getting really old, and this now abstracts the chain from callers of the methods on this object. We could probably move the signer here too.

Reply:
If you want to run with making these types stateful I'm also not opposed to wrapping actor state accessors like the new view up in these and expanding their functionality to do, for example, power table lookups.